src/katgpucbf/fgpu/kernels/postproc.mako

```diff
@@ -22,8 +22,21 @@
 #define OUT_CHANNELS ${out_high - out_low}
 #define n ${unzip_factor}
 #define m ${channels // unzip_factor}
+#define OUT_BITS ${out_bits}
+
+#if OUT_BITS == 4
+// quantised complex number
+typedef unsigned char qcomplex;
+// quantised Jones vector (pair of complex numbers)
+typedef uchar2 qjones;
+#define QMAX 7.0f
+#else
+typedef char2 qcomplex;
+typedef char4 qjones;
+#define QMAX 127.0f
+#endif
 
-<%transpose:transpose_data_class class_name="scratch_t" type="char4" block="${block}" vtx="${vtx}" vty="${vty}"/>
+<%transpose:transpose_data_class class_name="scratch_t" type="qjones" block="${block}" vtx="${vtx}" vty="${vty}"/>
```

Comment:
Not sure I'm going to be much use in this domain, but worth asking where I can find any kind of pointers/documentation on understanding this syntax (and it's uses)? (I assume it's Mako? Naive Google didn't turn up anything.)

Reply:
It's Mako. For the transposition macro specifically, https://katsdpsigproc.readthedocs.io/en/latest/user/macros.html is the reference.